lib/Sema/DerivedConformanceCodable.cpp

```diff
@@ -1053,14 +1109,12 @@ static ValueDecl *deriveDecodable_init(TypeChecker &tc, Decl *parentDecl,
   // Func name: init(from: Decoder)
   DeclName name(C, C.Id_init, paramList);
 
-  auto *initDecl = new (C) ConstructorDecl(
-      name, SourceLoc(),
-      /*Failability=*/OTK_None,
-      /*FailabilityLoc=*/SourceLoc(),
-      /*Throws=*/true, /*ThrowsLoc=*/SourceLoc(), selfDecl, paramList,
-      /*GenericParams=*/nullptr, target);
+  auto *initDecl = new (C) ConstructorDecl(name, SourceLoc(), OTK_None,
+                                           SourceLoc(), /*Throws=*/true,
+                                           SourceLoc(), selfDecl, paramList,
+                                           /*GenericParams=*/nullptr, target);
   initDecl->setImplicit();
-  initDecl->setBodySynthesizer(deriveBodyDecodable_init);
+  initDecl->setBody(deriveBodyDecodable_init(tc, initDecl));
```

Comment:
Unfortunately, I think this is unavoidable here. We need to pass the `TypeChecker` in for member lookups, and the lazy synthesizer version only takes functions which have one argument — the decl itself. We've run into this before, I think. The alternative is to get an `ASTContext` from the `initDecl` and cast its lazy resolver to a `TypeChecker`, but I'm not sure how safe that is...